docugami_langchain/agents/re_act_agent.py

```diff
@@ -111,13 +105,16 @@ class AgentState(TypedDict):
     intermediate_steps: Annotated[list[tuple[AgentAction, str]], operator.add]
 
 
-class ReActAgent(BaseRunnable[AgentState]):
+class ReActAgent(BaseDocugamiAgent[AgentState]):
     """
     Agent that implements simple agentic RAG using the ReAct prompt style.
     """
 
     tools: list[BaseTool] = []
 
+    iterations: int = 0
```

Comment:
iterations should be scoped in a run method if I'm getting their purpose correctly and not global to the Agent class